src/components/console/Console.cpp

```diff
@@ -0,0 +1,241 @@
+    if(hasCommandFlag)
+    {
+        hasCommandFlag = false;
+
+        static constexpr int maxArgumentsCount = 20;
+        static constexpr int maxBufferLength = 128;
+
+        char arg_buffer[maxBufferLength];
+        const char *args[maxArgumentsCount];
+        
+        // Copy string, becase we replace ' ' with '\0' for proper string termination
+        strncpy(arg_buffer, rxBuffer, sizeof(arg_buffer));
+
+        // First argument is always command name itself
+        uint16_t argc = 1;
+        args[0] = arg_buffer;
+
+        uint16_t param_len = strlen(rxBuffer);
+
+        for (uint8_t i = 0; i < param_len; i++)
+        {
+            if (rxBuffer[i] == ' ' && param_len > (i + 1))
+            {
+                arg_buffer[i] = '\0';
+                args[argc++] = &arg_buffer[i+1];
+            }
+
+            if (argc == maxArgumentsCount)
+            {
+                // Max argument count reached
+                break;
+            }
+        }
+
+        // This AT > OK needs to be there, because https://terminal.hardwario.com/ waits for the answer
+        // When we use or create better webpage terminal, this can go out
+        if(cmdCmp(rxBuffer, "AT"))
+        {
+            Print((char*)"OK\r\n");
+        }
+        else if(cmdCmp(rxBuffer, "COLOR"))
+        {
+            if(argc == 2)
+            {
+                setObjectsColor(args[1]);
+            }
+            else
+            {
+                Print("Expects 1 parameter");
+            }
+        }
+        else if(cmdCmp(rxBuffer, "LVGL"))
+        {
+            CommandLvgl(args, argc);
+        }
+        else if(cmdCmp(rxBuffer, "VIBRATE"))
+        {
+            motorController.SetDuration(100);
+        }
+        else if(cmdCmp(rxBuffer, "FS"))
+        {
+            // TODO: add directory listings etc.
+        }
+        else if(cmdCmp(rxBuffer, "WKUP"))
+        {
+            systemTask.PushMessage(Pinetime::System::Messages::GoToRunning);
+        }
+        else if(cmdCmp(rxBuffer, "SLEEP"))
+        {
+            systemTask.PushMessage(Pinetime::System::Messages::GoToSleep);
+        }
+        else if(cmdCmp(rxBuffer, "SPINOR"))
+        {
+            // TODO: print RAW data from FLASH
+        }
+        else if(cmdCmp(rxBuffer, "ACC"))
+        {
+            // Print 50 accelerometer measurements
+            accCount = 50;
+        }
+    }
+
+    // Debug print accelerometer values
+    if(accCount)
+    {
+        accCount--;
+        AccelerometerDebug();
+    }
+}
+
+void Console::Received(char* str, int length)
+{
+    for(int i = 0; i < length; i++)
+    {
+        // Wrap if input is too long without CR/LN
+        if(rxPos == bufferSize - 1)
+        {
+            rxPos = 0;
+        }
+
+        rxBuffer[rxPos++] = str[i];
+        rxBuffer[rxPos] = '\0'; // terminate for debug print 
+
+        if(str[i] == 13 || str[i] == 10)
```

Comment:
Using `'\n'` and `'\r'` would make this clearer.

Reply:
Not sure why I used numbers :) And I looked for them in ASCII table to have them right :facepalm:  a3317d8